pkg/apis/apps/v1alpha1/apimanager_types_test.go

```diff
@@ -0,0 +1,113 @@
+package v1alpha1
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/product"
+	"github.com/3scale/3scale-operator/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSetDefaults(t *testing.T) {
```

Comment:
The organization of the test file I've thought is the following one:
 * You have one test function (uppercase) per exported method of the file you want to test (in this case apimanager_types.go). In this function you have several subtests that you add with t.Run for each of them to test the different situations in that method that you want to check.
Another alternative would be to create one exported test function per situation you want to test per method you want to test. The downside of this is that you might end up with duplicated code and you would have more exported methods.

What do you think?